src/lib/files.js

```diff
@@ -87,42 +76,20 @@ export async function makeCIDFromFiles (files, ipfs) {
   return stat.cid
 }
 
-/**
- *
- * @param {FileStat[]} files
- * @param {string} apiUrl
- * @param {IPFSService} ipfs
- * @returns {Promise<FileDownload>}
- */
-async function downloadMultiple (files, apiUrl, ipfs) {
-  if (!apiUrl) {
-    const e = new Error('api url undefined')
-    return Promise.reject(e)
-  }
-
-  const cid = await makeCIDFromFiles(files, ipfs)
-
-  return {
-    url: `${apiUrl}/api/v0/get?arg=${cid}&archive=true&compress=true`,
-    filename: `download_${cid}.tar.gz`,
-    method: 'POST' // API is POST-only
-  }
-}
-
 /**
  *
  * @param {FileStat[]} files
  * @param {string} gatewayUrl
- * @param {string} apiUrl
  * @param {IPFSService} ipfs
- * @returns {Promise<FileDownload>}
+ * @returns {Promise<string>}
  */
-export async function getDownloadLink (files, gatewayUrl, apiUrl, ipfs) {
+export async function getDownloadLink (files, gatewayUrl, ipfs) {
   if (files.length === 1) {
-    return downloadSingle(files[0], gatewayUrl, apiUrl)
+    return getDownloadURL(files[0].type, files[0].name, files[0].cid, gatewayUrl)
   }
 
-  return downloadMultiple(files, apiUrl, ipfs)
+  const cid = await makeCIDFromFiles(files, ipfs)
+  return getDownloadURL('directory', '', cid, gatewayUrl)
```

Comment:
@hacdias passing empty `name` here will always produce `download_<cid>.tar`.
It has been a while, do you remember if there is any reason why we can't preserve directory name and do something like `<dir_name>_<cid>.tar` ?